extensions/amp-date-picker/1.0/component/date-range-picker.tsx

```diff
@@ -0,0 +1,432 @@
+
+      const isAfterDisabledDate =
+        disabledAfter && endDate && isAfter(endDate, disabledAfter);
+      if (isAfterDisabledDate) {
+        return;
+      }
+
+      const isBeforeDisabledDate =
+        disabledBefore &&
+        endDate &&
+        !isSame &&
+        isBefore(endDate, disabledBefore);
+      if (isBeforeDisabledDate) {
+        return;
+      }
+
+      if (focusedInput === 'start-input') {
+        handleSetStartDate(startDate);
+      } else if (focusedInput === 'end-input') {
+        handleSetEndDate(endDate);
+      }
+
+      if (isSame) {
+        setFocusedInput('end-input');
+      }
+
+      if (isFinalSelection) {
+        transitionTo('overlay-closed');
+      }
+    },
+    [
+      handleSetStartDate,
+      handleSetEndDate,
+      isBlockedRange,
+      getDisabledAfter,
+      getDisabledBefore,
+      focusedInput,
+      transitionTo,
+      openAfterSelect,
+    ]
+  );
+
+  /**
+   * Formats date range for react-day-picker
+   */
+  const dateRange = useMemo<DateRange>(() => {
+    return {
+      from: startDateInput.date,
+      to: endDateInput.date,
+    };
+  }, [startDateInput, endDateInput]);
+
+  /**
+   * Resets the date and input value and returns to the initial month
+   */
+  const clear = useCallback(() => {
+    setMonth(defaultMonth);
+    startDateInput.clear();
+    endDateInput.clear();
+
+    if (openAfterClear) {
+      transitionTo('overlay-open-input');
+    }
+  }, [
+    defaultMonth,
+    startDateInput,
+    endDateInput,
+    openAfterClear,
+    transitionTo,
+  ]);
+
+  useImperativeHandle(
+    ref,
+    () => ({
+      clear,
+      setDates: (startDate: Date, endDate: Date) => {
+        // Note: This function does not set the month or respect blocked dates
+        // to maintain feature parity with the original amp date picker.
+        handleSetStartDate(startDate);
+        handleSetEndDate(endDate);
+      },
+      startToday: startDateInput.setToToday,
+      endToday: endDateInput.setToToday,
+    }),
+    [clear, startDateInput, endDateInput, handleSetStartDate, handleSetEndDate]
+  );
+
+  /**
+   * For inputs that are valid dates, update the date-picker value.
+   */
+  const handleStartInput = useCallback(
+    (e: InputEvent) => {
+      const {target} = e;
+      if ((target as HTMLInputElement).type === 'hidden') {
+        return;
+      }
+
+      const date = parseDate((target as HTMLInputElement).value);
+      if (date && isValid(date)) {
+        startDateInput.setDate(date);
+      }
+    },
+    [startDateInput, parseDate]
+  );
+
+  /**
+   * For inputs that are valid dates, update the date-picker value.
+   */
+  const handleEndInput = useCallback(
+    (e: InputEvent) => {
+      const {target} = e;
+      if ((target as HTMLInputElement).type === 'hidden') {
+        return;
+      }
+
+      const date = parseDate((target as HTMLInputElement).value);
+      if (date && isValid(date)) {
+        endDateInput.setDate(date);
+      }
```

Comment:
Same as above - I personally think it is simpler to keep these separate rather than adding another argument and a conditional, but that's just my opinion!